src/components/policy/policy_external/src/update_status_manager.cc

```diff
@@ -184,14 +190,20 @@ void UpdateStatusManager::DoTransition() {
 
   current_status_ = next_status_;
   next_status_.reset();
-  listener_->OnUpdateStatusChanged(current_status_->get_status_string());
+
+  LOG4CXX_DEBUG(logger_, "last_processed_event_ = " << last_processed_event_);
+  if (last_processed_event_ != kScheduleManualUpdate) {
+    listener_->OnUpdateStatusChanged(current_status_->get_status_string());
+  }
```

Comment:
The one difference I see with this solution is that this line will no longer be called:
```
policy_manager_->SaveUpdateStatusRequired(policy::kUpToDate != status);
```
Would it cause any problems not updating the policy table's `update_needed` status?